external/storm-hdfs-blobstore/src/main/java/org/apache/storm/hdfs/blobstore/HdfsBlobStoreImpl.java

```diff
@@ -312,4 +313,54 @@ public void shutdown() {
             timer.cancel();
         }
     }
+
+    /**
+     * Get the last modification time of any blob.
+     *
+     * @return the last modification time of blobs within the blobstore.
+     * @throws IOException on any error
+     */
+    public long getLastModTime() throws IOException {
+        long modtime =  fileSystem.getFileStatus(fullPath).getModificationTime();
+        return modtime;
+    }
+
+    /**
+     * Updates the modification time of the blobstore to the current time.
+     *
+     * @throws IOException on any error
+     */
+    public void updateLastModTime() throws IOException {
+        long timestamp = Time.currentTimeMillis();
+        fileSystem.setTimes(fullPath, timestamp, timestamp);
```

Comment:
I am not very sure if changing the mtime of the blobstore directory is a good idea. It breaks the semantics of modification time of a directory.

Reply:
I am not clear what the concern is.  Would you prefer I touch a file instead of directory?  Other suggestions?

Reply:
I would personally prefer to use a separate file to record the metadata instead of modifying the mtime of a directory since it doesn't follow the semantics of mtime. It doesn't seem to be a problem currently but we don't know what we will do in the future so I'd rather not break the semantics and also this is to avoid confusion. It is a personal preference though.

Reply:
will update the PR with your suggested change.

Reply:
Updated with this change and addressed your other comments.